### DISCUSSION TITLE
Why my clangd (vscode) can find c standard head files but can't find c++ standard head files (like iostream)?

### DISCUSSION THREAD

**Discussion by zhiqiangzz** [clangd/clangd#1665] [Q&A] [answered]
```shell
# uname -a
Linux ubuntu-linux-22-04-desktop 5.15.0-73-generic #80-Ubuntu SMP Mon May 15 15:18:50 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

# ls -l /usr/bin | grep g++
aarch64-linux-gnu-g++ -> g++-11
aarch64-linux-gnu-g++-11
g++ -> g++-11
g++-11 -> aarch64-linux-gnu-g++-11

# clangd --version
Ubuntu clangd version 14.0.0-1ubuntu1
```

I use gcc/g++ as my compiler.
command `gcc -v -c -xc++ /dev/null` shows 

```shell
#include <...> search starts here:
 /usr/include/c++/11
 /usr/include/aarch64-linux-gnu/c++/11
 /usr/include/c++/11/backward
 /usr/lib/gcc/aarch64-linux-gnu/11/include
 /usr/local/include
 /usr/include/aarch64-linux-gnu
 /usr/include
```

But my c file can detect <stdio.h> and I can get into <stdio.h> to get infos in vscode. vscode smart tip words well.

But when I create a cpp file and include <iostream> or other c++ standard head files, my vscode can't detect.

when I put follow  to my ~/.clangd/config.yaml

```shell
CompileFlags:
  Add: [-isystem,/usr/include/c++/11]
```

ok,my vscode can detect the c++ standard head files but vscode still show many errors such as can't find std::cout .

when I put follow to my .vscode/settings.json and through cmake .. to build the project , the /build directory generate compile_commands.json.

```shell
"clangd.arguments": [
        "--query-driver=/usr/bin/aarch64-linux-gnu-g++,/usr/bin/aarch64-linux-gnu-gcc",
    ],
```

all cpp files in current project can find c++ standard head files and vscode smart tips can work but some head files can't find c++ standard head files.

and  in my env must be --query-driver=/usr/bin/aarch64-linux-gnu-g++ but not just --query-driver=g++ the cmake builded project can work. Though I find my g++ slink to /usr/bin/aarch64-linux-gnu-g++.

many thanks!

**Comment by HighCommander4** [comment, marked as answer]
You're likely running into https://github.com/clangd/clangd/issues/1394, please see the [workarounds](https://github.com/clangd/clangd/issues/1394#issuecomment-1328676884) discussed there (with the most straightforward one being installing `g++-12`)

**Reply by zhiqiangzz** [reply]
pretty thanks it words !!!!!!! you give me a big favor!
